appinventor/appengine/src/com/google/appinventor/client/editor/simple/components/MockForm.java

```diff
@@ -1351,9 +1353,14 @@ public TreeItem buildComponentsTree() {
    * @return  tree showing the component hierarchy of the form
    */
   public TreeItem buildComponentsTree(Integer view) {
+    this.view = view;
     return buildTree(view);
   }
 
+  public Integer getView() {
```

Comment:
This does not appear to be used.

Could you talk through the reasoning on introducing a new view attribute?

Reply:
So, there was another issue: when you drag and drop an element to the Mock, the state of the tree is not preserved. For example, if you were viewing only Visible components, the tree will reset to the All components on any tree change.

This `view` variable preserves the selected view state on a per-Form level, avoiding this unintended behavior.

Reply:
OK, that makes sense. the getter here doesn't appear to be called anywhere. Is that correct? I think we can skip adding a getter until we need access the attribute outside the class.

Reply:
Resolved. It was added as I was running some tests, but didn't achieve anything (fixing the Known Issue mentioned above).